packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/OutboundLineEdit.tsx

```diff
@@ -110,14 +111,20 @@ export const OutboundLineEdit: React.FC<ItemDetailsModalProps> = ({
           allocatedQuantity={getAllocatedQuantity(draftOutboundLines)}
           availableQuantity={sumAvailableQuantity(draftOutboundLines)}
           onChangeQuantity={onAllocate}
+          canAutoAllocate={!!(currentItem && draftOutboundLines.length)}
```

Comment:
this is now a big conditional mess.

Wondering what to do about the finalised state of the modal and what to do there. I imagine that will be a significant change also, so thought to leave it messy for now.

Reply:
a small suggestion is to pull the duplicated `!!(currentItem && draftOutboundLines.length)` out to a variable (`canAutoAllocate`?) just to help readability